### PR TITLE
add external network apis

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This PR adds the required resources needed for the new External Network.

### CODE REVIEW

File: apis/networking/v1alpha1/configuration_types.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	// CIDR of the cluster.
+	CIDR CIDR `json:"cidr,omitempty"`
+}
+
+// ConfigurationSpec defines the desired state of Configuration.
+type ConfigurationSpec struct {
+	// Local network configuration (the cluster where the resource is created).
+	Local ClusterConfig `json:"local,omitempty"`
+	// LocalRemapped network configuration, it defines how the remote cluster sees the local cluster.
+	LocalRemapped ClusterConfig `json:"localRemapped,omitempty"`
```
Comment (cheina97) [inline apis/networking/v1alpha1/configuration_types.go:56]: It should be optional, could be an idea to convert all ClusterConfig in pointers?

File: apis/networking/v1alpha1/gatewayserver_types.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+// EndpointStatus defines the observed state of the endpoint.
+type EndpointStatus struct {
+	// Addresses specifies the addresses of the endpoint.
+	Addresses []string `json:"addresses,omitempty"`
+	// Port specifies the port of the endpoint.
+	Port int32 `json:"port,omitempty"`
+	// Protocol specifies the protocol of the endpoint.
+	// +kubebuilder:validation:Enum=TCP;UDP
+	Protocol *corev1.Protocol `json:"protocol,omitempty"`
```
Comment (fra98) [inline apis/networking/v1alpha1/gatewayserver_types.go:66]: Does the enum validation works also with pointers?

File: apis/networking/v1alpha1/configuration_types.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ConfigurationResource the name of the configuration resources.
+var ConfigurationResource = "configuration"
```
Comment (fra98) [inline apis/networking/v1alpha1/configuration_types.go:26]: Should we define all resource names with the plural for coerence with rest of the codebase?

File: apis/networking/v1alpha1/connection_types.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	// ConnectionTypeServer represents a server connection.
+	ConnectionTypeServer ConnectionType = "Server"
+	// ConnectionTypeClient represents a client connection.
+	ConnectionTypeClient ConnectionType = "Client"
+)
+
+// PingSpec defines the desired state of Ping.
+type PingSpec struct {
+	// Enabled specifies whether the ping is enabled or not.
+	// +kubebuilder:default=true
```
Comment (fra98) [inline apis/networking/v1alpha1/connection_types.go:51]: Same question about pointer validation here

File: apis/networking/v1alpha1/connection_types.go [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+	ConnectionConditionStatusFalse ConnectionConditionStatusType = "False"
+	// ConnectionConditionStatusUnknown represents a connection condition that is unknown.
+	ConnectionConditionStatusUnknown ConnectionConditionStatusType = "Unknown"
+)
+
+// ConnectionCondition contains details about state of the connection.
+type ConnectionCondition struct {
+	// Type of the connection condition.
+	// +kubebuilder:validation:Enum="Established";"Pending";"Denied";"Error"
+	Type ConnectionConditionType `json:"type"`
```
Comment (fra98) [inline apis/networking/v1alpha1/connection_types.go:98]:
```suggestion
	// +kubebuilder:validation:Enum="Established";"Pending";"Denied";"Error"
    // +kubebuilder:default="Pending"
	Type ConnectionConditionType `json:"type"`
```

**Pull request merged**